discord/utils.py

```diff
@@ -151,7 +150,7 @@ def __get__(self, instance, owner):
     class _RequestLike(Protocol):
         headers: Mapping[str, Any]
 
-    cached_property = NewType("cached_property", property)
+    cached_property = functools.cached_property
```

Comment:
In my opinion, it feels weird to replace it with `functools.cached_property` only in `TYPE_CHECKING`. This means that when type checking the ide will think that `cached_property` has all the extra features of `functools.cached_property`. Maybe it could be better to actually replace the runtime imlementation with `functools.cached_property` (would need to be tested), and remove it from the `TYPE_CHECKING` logic entirely.

Reply:
That's understandable, I personally found this to work well, and I'm not sure how I would implement your idea, which is why I did it this way and think that this is fine.
If I find an even better way in the future I will follow up on it.